game/ato/flightplans/shiprecoverytanker.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Iterator, Type
+from game.ato.flightplans.ibuilder import IBuilder
+from .patrolling import PatrollingLayout
+from .refuelingflightplan import RefuelingFlightPlan
+from game.ato.flightplans.waypointbuilder import WaypointBuilder
+from game.ato.flightwaypoint import FlightWaypoint
+
+
+class RecoveryTankerFlightPlan(RefuelingFlightPlan):
+    @staticmethod
+    def builder_type() -> Type[Builder]:
+        return Builder
+
+    @property
+    def patrol_duration(self) -> timedelta:
+        return self.flight.coalition.game.settings.desired_tanker_on_station_time
+
+
+class Builder(IBuilder[RecoveryTankerFlightPlan, PatrollingLayout]):
+    def layout(self) -> PatrollingLayout:
+
+        builder = WaypointBuilder(self.flight)
+        # TODO: Propagate the ship position to the Tanker's TOT,
+        # so that we minimize the tanker's need to catch up to the carrier.
```

Comment:
I'm not entirely sure that we have access to it, but if we could reference the wind direction here that would help us fine-tune the position of the waypoint. Will try to check what possibilities we have tomorrow 🤞

Reply:
We have the wind-direction at our disposal, but we'll also need to TOT for an estimated location of the carrier. However, the TOT is currently a bit of a mess because it doesn't take into account when other flight start returning to the carrier, so that'd be the first thing to fix on my end for the auto-planner to work decently... yes, despite what I initially said about holding off on that I'm actually looking into it 😅